### PR TITLE
Add numpy copy function and its testing method

### LINKED ISSUE #5801: copy

### PR DESCRIPTION
**Pull request by farid002**
Close #5801

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_creation_routines/test_from_existing_data.py [hunk 1]
```diff
@@ -39,3 +39,38 @@ def test_numpy_asarray(
...
+    helpers.test_frontend_function(
+        input_dtypes=dtype,
+        as_variable_flags=as_variable,
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        frontend="numpy",
+        fn_tree="copy",
+        a=a,
+        dtype=dtype,
```
Comment (Daniel4078) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_creation_routines/test_from_existing_data.py:75]: there is no argument called "dtype" in the copy function, then why do you pass it in?

Comment (farid002) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_creation_routines/test_from_existing_data.py:75]: @Daniel4078 totally agree. However, surprisingly it works when I pass `dtype=dtype`. If I dont pass it in, I got a library error :
`**Could not load dynamic library 'libcuda.so.1'; dlerror: libcuda.so.1: cannot open shared object file: No such file or directory**`

Comment (Daniel4078) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_creation_routines/test_from_existing_data.py:75]: ok then

**Review by Daniel4078 — APPROVED** [review]
It seems like all fine to me for now. Well done.

**Pull request merged**